Airport_Ticket_Booking_System/Services/BookingService.cs

```diff
@@ -86,12 +87,11 @@ public void ModifyBooking(string bookingId, Flight newFlight, string newClassTyp
             return;
         }
 
-        booking = booking with
-        {
-            FlightNumber = newFlight.FlightNumber,
-            ClassType = newClassType,
-            Price = newFlight.Prices[newClassType]
-        };
+        booking = booking.modifyBooking(
+            flightNumber: newFlight.FlightNumber,
+            classType: newClassType,
+            flightPrice: newFlight.Prices[newClassType]
+        );
```

Comment:
I don't think the modifyBooking method is necessary.
You can move the logic directly here and remove it from the class.
Keeping everything encapsulated in this context is safer, since exposing modifyBooking allows it to be called from anywhere.